strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/LoggingManagementController.java

```diff
@@ -238,11 +237,111 @@ public ResponseEntity uploadLogbackConfiguration(HttpServletRequest request,
         catch (IOException | LoggingConfigurationException e)
         {
             String message = "Could not upload logback configuration.";
+            
             logger.error(message, e);
 
-            return ResponseEntity.status(BAD_REQUEST)
-                                 .body(getResponseEntityBody(message, accept));
+            return ResponseEntity.status(BAD_REQUEST).body(getResponseEntityBody(message, accept));
         }
     }
 
+    @ApiOperation(value = "Used to get log directory.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The log directory was retrieved successfully."),
+                            @ApiResponse(code = 400, message = "Could not download log directory.") })
+    @GetMapping(value = { "/logs",
+                          "/logs/**" },
+                produces = TEXT_PLAIN_VALUE)
+    public void generateLogDirectoryListing(HttpServletRequest request,
+                                            HttpServletResponse response)
+            throws IOException
+    {
+        String requestUriString = request.getRequestURI();
+        String uriLogDirPath = requestUriString.replace("/logging", "");
+        Path localLogDirPath = Paths.get(PropertyUtils.getVaultDirectory(), uriLogDirPath);
+        
+        if (Files.notExists(localLogDirPath))
+        {
+            response.sendError(404, "File not found. Variables: " + uriLogDirPath + " " + localLogDirPath.toString());
+        }
+        
+        File file = localLogDirPath.toFile();
+        
+        //Sends a redirect if the URI does not end with a "/"
+        if (!requestUriString.endsWith("/"))
+        {
+            try
+            {
+                response.sendRedirect(requestUriString + "/");
+            }
+            catch (IOException e)
+            {
+                logger.debug("Error redirecting to " + requestUriString + "/");
+            }
```

Comment:
You need to watch out for the `return` after the `try/catch` block here as well. If don't `return` - it will send a `redirect` response to the client and will continue executing the rest of the code.

Reply:
Fixed in the latest commit!